[HOLD] Implement Uneven Splits

youssef-lr opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

- [x] Hold on https://github.com/Expensify/Auth/pull/10709
- [x] Hold on  https://github.com/Expensify/Web-Expensify/pull/41839

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/40616



### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### Creating an uneven split from Global Create

1. Click on the ‘+’ button.
2. Choose ‘Split expense’
3. Enter an amount
4. Type in a few participants and press ‘split’ on each of them.
5. Confirm the selection
6. Click on ‘Amount’ and change it.
7. Make sure we have adjusted the even shares based on the new amount.
8. Change the amount of the first participant.
9. Verify the amounts of the remaining participants have been adjusted and are equal to even splits of the remaining total (total - amount you entered)
10. Repeat the same step for each participant until the last participant has automatically the remaining correct amount.
11. Confirm the split.
12. Verify the money request reports have been created and have the correct amounts.
13. Click on the Split preview card
14. Verify the individual amounts are correctly shown next to each participant.

### Creating an uneven split from a group chat

1. Create a group chat with a few participants
2. Click the ‘+’ button from the group chat and choose ‘Split Bill’
3. Enter an amount
4. Change the amount of the participants in the inputs next to each participant.
5. Confirm the split.
6. Verify the money request reports have been created and have the correct amounts.
7. Click on the Split preview card
8. Verify the correct amounts are shown next to each participant.

### Error handling

1. Click on the ‘+’ button.
2. Choose ‘Request Money’
3. Enter an amount
4. Type in a few participants and press ‘split’ on each of them.
5. Confirm the selection
6. Change the amount of the first participant, choose an amount that exceeds the total value of the bill.
7. Verify an error message is shown that says “You enter <amount> but the total is <amount>”
8. Verify clicking on ‘Split’ does nothing.
9. Edit the amount so that it’s correct
10. Verify the error message disappears.
11. Confirm the split and make sure it gets created properly.

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

1. Go offline
2. Click on the ‘+’ button.
3. Choose ‘Request Money’
4. Enter an amount
5. Type in a few participants and press ‘split’ on each of them and confirm the selection.
6. Adjust the amounts next to each participant using correct values.
7. Create the split.
8. Verify the split is created offline
9. Verify the preview card shows the correct participants.
10. Click on the preview to view the split details page.
11. Verify the page shows the correct amounts next to each participant.
12. Go back online. 
13. Verify the split action is no longer grayed out
14. Verify individual chats and money reports have the correct amounts.


### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.


### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

youssef-lr committed:
Add TS definitions

youssef-lr committed:
Set split shares when first creating the split

youssef-lr committed:
Add IOU functions to set split shares and adjust them

youssef-lr committed:
Allow configuring split amounts

youssef-lr committed:
Use temporary amount input

youssef-lr committed:
Pass custom styles

youssef-lr commented on src/components/MoneyRequestAmountInput.tsx:
```diff
@@ -0,0 +1,206 @@
+import React, {useCallback, useEffect, useRef, useState} from 'react';
```
For now I copied and pasted code from `MoneyRequestAmountForm`, we're working on making the underlying amount input reusable here https://github.com/Expensify/App/issues/40382

youssef-lr committed:
Temporary updates to the amount input

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
Allow setting individual shares

youssef-lr committed:
Show a form error if sum of shares don't match total

youssef-lr committed:
Prevent submitting form if total doesn't match

youssef-lr committed:
Adjust shares automatically

youssef-lr committed:
Revert changes

youssef-lr committed:
Use temporary input

youssef-lr committed:
Show even splits for workspace chat splits

youssef-lr committed:
Send uneven splits to backend

youssef-lr committed:
Save splits array in the transaction optimistically

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
Copy changes to MoneyRequestConfirmationList

youssef-lr committed:
Allow excluding group chat users from splits by setting amount to 0

youssef-lr committed:
Reset split shares when amount changes

youssef-lr committed:
Display read only shares in splti details page

youssef-lr committed:
Show split share in split preview card

youssef-lr committed:
Allow easy tab navigation for inputs

youssef-lr committed:
Refactoring and TS fixes

youssef-lr committed:
wip: Reset button

youssef-lr committed:
Allow resetting the amounts

youssef-lr committed:
Bug fix

youssef-lr committed:
Debounce form error in splits page

youssef-lr committed:
Bug fix

youssef-lr committed:
Lint

youssef-lr committed:
Hide inputs for Smartscan splits

youssef-lr commented:
Working on conflicts. This isn't 100% ready for review, what's remaining is:

1. Waiting for this PR so we can reuse the amount input instead of the copy and pasted component: reusable and stylable money request amount input by ZhenjaHorbach · Pull Request #40390 · Expensify/App
2. Fixing a flicker issue in the amounts due to this https://github.com/Expensify/App/pull/40386/files#diff-9c07b44e092fd3d975806649be46d76b75a377c16e32ed86597b630a84c290d0R5864
3. Spanish copy

But the rest of functionality can be reviewed and tested now

c3024 commented:
QAB flow is broken for splitting bill after splitting with an uneven amount
![Screenshot 2024-04-23 at 6 34 43 AM](https://github.com/Expensify/App/assets/102477862/c8011dca-7100-4eb9-adbe-394f1dae50b6)

Currency and amount are not aligned
![Screenshot 2024-04-23 at 6 35 22 AM](https://github.com/Expensify/App/assets/102477862/190b6152-d46d-4781-b514-85694532859c)

Split preview card on clicking the report action shows a currency different from the split currency.
![Screenshot 2024-04-23 at 6 35 34 AM](https://github.com/Expensify/App/assets/102477862/7ed7f235-b045-45ab-ba6f-c24e841639ef)

youssef-lr commented:
>Currency and amount are not aligned

Yeah this will be fixed once the PR I mentioned in my last comment is merged.

youssef-lr committed:
Fix currency not shown in split amounts in details page

youssef-lr commented:
>Currency and amount are not aligned

Fixed.

Looking into "QAB flow is broken for splitting bill after splitting with an uneven amount"

youssef-lr commented:
Also another known issue is if you sign out then back in the splits will be show as even shares. This will be fixed once we start saving the splits array in the transaction in the backend.

youssef-lr committed:
Bug fix

youssef-lr committed:
Filter out participants with shares equal to 0

youssef-lr committed:
Prevent creating a split containing a single participant

youssef-lr committed:
Prevent submitting form if there's an error

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
Lint

youssef-lr commented on src/pages/iou/request/step/IOURequestStepConfirmation.tsx:
```diff
@@ -524,12 +535,6 @@ function IOURequestStepConfirmation({
-                        // but not all of them (maybe someone skipped out on dinner). Then it's nice to be able to select/deselect people from the group chat bill
-                        // split rather than forcing the user to create a new group, just for that expense. The reportID is empty, when the action was initiated from
-                        // the floating-action-button (since it is something that exists outside the context of a report).
-                        canModifyParticipants={!transaction?.isFromGlobalCreate}
```
We're removing the option to select/deselect participants when the split is initiated from a GC

c3024 commented on src/components/MoneyRequestConfirmationList.tsx:
```diff
@@ -413,43 +401,99 @@ function MoneyRequestConfirmationList({
-                },
-            );
+            const formattedParticipantsList = getParticipantOptions();
+            sections.push({
```
![Screenshot 2024-04-23 at 1 48 15 PM](https://github.com/Expensify/App/assets/102477862/1c8f0ac1-dd5c-49b5-919c-73652572d2dd)

This PR on left. `main` on right.

'Paid by' section is removed. Is this intentional?

Could you share the section of the document for this feature on kkch3024@gmail.com if it is allowed? I have signed the NDA.

The amounts are missing for some participants.

youssef-lr replied:
>Also, the amounts are missing for some participants.

Unable to reproduce this one either.

youssef-lr replied:
Actually I can now with Hidden accounts

youssef-lr replied:
This will be fixed in the backend. Because the accountIDs used in that page are real accountIDs which should come back from the HTTPS response

youssef-lr replied:
>'Paid by' section is removed. Is this intentional?

Yes, it will be brought back with this [PR](https://github.com/Expensify/App/pull/40388)

c3024 commented:
Starting a split bill from group chat, the sum of amounts do not match the split total amount

https://github.com/Expensify/App/assets/102477862/5be5b10e-138d-42ec-bb33-e8083266e559

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
Add margin next to participant emails

youssef-lr commented:
Thanks @c3024, taking a look tonight

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
Always show confirmation for uneven splits

youssef-lr commented:
Fixed the QAB issue, we have to always go to the confirmation page for uneven splits.

youssef-lr committed:
Fix input style

youssef-lr commented:
>Starting a split bill from group chat, the sum of amounts do not match the split total amount

I'm unable to reproduce this one..

youssef-lr committed:
Clean up code for section button

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
TS lint

youssef-lr committed:
More lint

youssef-lr commented on src/pages/iou/request/step/IOURequestStepAmount.tsx:
```diff
@@ -181,25 +181,7 @@ function IOURequestStepAmount({
-                    });
-                    return;
-                }
+            if (shouldSkipConfirmation && iouType !== CONST.IOU.TYPE.SPLIT) {
```
It doesn't make sense to skip the confirmation screen for the case of uneven splits. Because each split can be configured now. Though I think I should bring this back for Smartscan & Workspace splits.

youssef-lr committed:
Skip confirmation screen for smartscanned splits

youssef-lr committed:
Remove workaround for removing participants

youssef-lr commented:
@c3024 this should be ready for another review please

youssef-lr committed:
Bring back effect

youssef-lr committed:
Lint

youssef-lr committed:
Don't run unneeded logic unless we have modified shares

youssef-lr committed:
Translation

youssef-lr committed:
TS

youssef-lr committed:
More TS

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
TS and comment imporvement

youssef-lr committed:
Remove unintended changes

youssef-lr committed:
One last TS fix

c3024 commented:
> > Starting a split bill from group chat, the sum of amounts do not match the split total amount
> 
> I'm unable to reproduce this one..

Rechecked. This is fine for already known users. This issue occurs for unknown users for whom backend returns Hidden.

dubielzyk-expensify commented:
The reset button shows up even when the values aren't tweaked. The reset button should only appear once you've changed one of the split values

![CleanShot 2024-04-26 at 13 16 16@2x](https://github.com/Expensify/App/assets/145416827/dca99767-7ef6-40a7-8233-5aa7caf2b8f3)

youssef-lr commented:
Good catch @dubielzyk-expensify, will update!

youssef-lr commented:
> Rechecked. This is fine for already known users. This issue occurs for unknown users for whom backend returns Hidden.

Yes this should be fixed in the backend where we'll save the real accountIDs in `transaction.comment.splits`.

youssef-lr commented:
@dubielzyk-expensify it's being worked on here, we're probably gonna merge it today. 

@c3024 please help test this we really would like to have it merged today as well.

dubielzyk-expensify commented:
This looks pretty good to me. Keen to get @Expensify/design eyes as well.

c3024 commented on src/components/MoneyRequestConfirmationList.tsx:
```diff
@@ -979,19 +1038,17 @@ function MoneyRequestConfirmationList({
             shouldUseStyleForChildren={false}
-            optionHoveredStyle={canModifyParticipants ? styles.hoveredComponentBG : {}}
-            footerContent={footerContent}
+            footerContent={!isEditingSplitBill && footerContent}
```
Is this intentional?

Previously after updating amount (for a scan request while scanning) and clicking the footer button the amount cannot be updated anymore.

With the button removed, now the amount can be changed multiple times and scanning also does not stop with updating the amount.

https://github.com/Expensify/App/assets/102477862/7b62efe4-1dd8-4624-9dc6-9e1e5554b6fa

First few seconds of the video is for this PR and remaining part on staging.

youssef-lr replied:
I have no idea how this change landed in my PR 😅 definitely a mistake. Thanks for catching this!

youssef-lr committed:
Remove unintended change

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
Cleanup

youssef-lr committed:
Cleanup

youssef-lr committed:
Header new copy

youssef-lr committed:
Add comment

c3024 commented on src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:
```diff
@@ -211,6 +211,16 @@ function MoneyRequestPreviewContent({
+        () =>
+            shouldShowSplitShare &&
+            (transaction?.comment?.splits
+                ? transaction?.comment?.splits?.find((split) => split.accountID === action.actorAccountID)?.amount
```
As of now I see equal share for the other user [as expected in this comment](https://github.com/Expensify/App/pull/40386#issuecomment-2071234655).

But here the actorAccountID for the action is the one who creates the split, right? I think we should use here currentUserAccountID.

youssef-lr replied:
Totally

youssef-lr committed:
Show current user's share in the split preview

c3024 commented:
Code changes look good. Testing again to complete the checklist.

c3024 commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

c3024 commented:
I am getting [this continuous loading/blank report page](https://expensify.slack.com/archives/C049HHMV9SM/p1714081597264369) when checking if the report preview of individual chats is working well.

c3024 commented:
Android build has been throwing errors on my device. Tried a few times already. Could you run a build here?

c3024 commented:
The 'merchant empty' error disappears immediately after appearing. Works well on staging.


https://github.com/Expensify/App/assets/102477862/551be404-d2fa-4179-8bda-081c2e2ddf79

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
Fix form error disappearing

youssef-lr commented:
Fixed @c3024. Since I switched to use `useDebouncedState`, the `setFormError` had to be included in the dependencies of this effect. I removed it and added a lint exception, it seems working fine. Not sure why the lint complains for `setForm` taht's coming fro `useDebouncedState` but not when it's coming from `useState`.

youssef-lr commented:
Investigating merchant field error not showing

<img width="397" alt="Screenshot 2024-04-26 at 16 53 07" src="https://github.com/Expensify/App/assets/9680864/36e0c8fd-f2b1-4131-a171-007e1be086d7">

youssef-lr committed:
Fix merchant error missing when completing a split bill

youssef-lr commented:
Ok fixed

youssef-lr committed:
Cleanup

c3024 reviewed (approved):
LGTM

youssef-lr commented:
@dangrous this is ready for review but not merging. I'm making a one liner PR in Auth that should be deployed first then we can merge.

rafecolton reviewed (commented):
Looks like we are displaying USD amounts in a non-standard way

![Screenshot 2024-04-26 at 10 06 36 AM](https://github.com/Expensify/App/assets/1058475/95205f11-7346-46e8-9839-3fc53b7ca55f)

When the amounts are edited, we're also not showing the right amount of decimals

![Screenshot 2024-04-26 at 10 12 11 AM](https://github.com/Expensify/App/assets/1058475/958d3284-e986-4772-8c04-1c0ba908ca8a)

youssef-lr commented:
Note: the bugs in the split page when navigating to the participant profile or their split share not showing will be fixed once backend is deployed. The reason for that is we only have their optimistic accountIDs stored in the `splits` prop of the transaction. The backend should return their actual accountIDs and the splits prop will be updated.

dangrous commented on src/libs/actions/IOU.ts:
```diff
@@ -5809,6 +5833,104 @@ function setShownHoldUseExplanation() {
+    const participantAccountIDsWithoutCurrentUser = newAccountIDs.filter((accountID) => accountID !== userAccountID);
+
+    // Create an array containing unique IDs of the current transaction participants and the new ones
+    const accountIDs = [...new Set<number>([userAccountID, ...participantAccountIDsWithoutCurrentUser, ...oldAccountIDs])];
```
NAB - can't we just do `[...newAccountIDs, ...oldAccountIDs]` instead of adding back in the userAccountID we just removed?

youssef-lr replied:
Thing is, if this is called from the participants selector step, we won't have current user account ID in newAccountIDs, if it's called from a split created from a group chat, it will have it included :/

youssef-lr replied:
So to be safe we just make sure to always remove it and add it back. Curious if you can see how to make this better!

dangrous replied:
ohh i see. Hm, yeah then this makes sense then

youssef-lr replied:
Will add a comment

youssef-lr replied:
I just thought of an improvement to the code too

youssef-lr replied:
Updated!

dangrous commented on src/libs/actions/IOU.ts:
```diff
@@ -5809,6 +5833,104 @@ function setShownHoldUseExplanation() {
+        .map((key: string): number => transaction?.splitShares?.[Number(key)]?.amount ?? 0)
+        .reduce((prev: number, current: number): number => prev + current, 0);
+
+    const unModifiedSharesAccountIDs = Object.keys(transaction.splitShares ?? {})
```
NAB
```suggestion
    const unmodifiedSharesAccountIDs = Object.keys(transaction.splitShares ?? {})
```
since it's one word

dangrous reviewed (dismissed):
I'm provisionally approving this, but also noting the same formatting issues that @rafecolton did, as well as some NABs in the code. But looks good overall so if we're okay with those weirdnesses, we can move forward (in my view!)

youssef-lr committed:
Rename variable

youssef-lr commented:
>but also noting the same formatting issues that @rafecolton did, as well as some NABs in the code. But looks good overall so if we're okay with those weirdnesses, we can move forward (in my view!)

Working on pushing an update for that right now!

youssef-lr committed:
Clean up code

youssef-lr committed:
Fix selection always moving to the end

youssef-lr committed:
Proper fix for selection moving when amount changes

youssef-lr committed:
Add comment

youssef-lr committed:
Clarify comment

youssef-lr committed:
Copy update

youssef-lr committed:
Clean up code

youssef-lr committed:
Format amounts

youssef-lr committed:
Cleanup

youssef-lr commented:
@c3024 could you check this [commit](https://github.com/Expensify/App/pull/40386/commits/14d577590a81e20fb0209a290c3e5a19a6b4689f) please? it implements this change where we'll display formatted amounts


https://github.com/Expensify/App/assets/9680864/4a4faddc-630e-4883-b917-aad53d95ee60

dubielzyk-expensify commented:
See there's a bit of activity here. Just tag design when you want another review here :)

youssef-lr commented:
This should be ready for a final testing round!

youssef-lr commented:
I noticed the keyboard is disabled on native. Pushing a fix now.

c3024 commented:
Tests well on web chrome


https://github.com/Expensify/App/assets/102477862/07168fbb-f1c1-46d1-8c2a-300bb6ce4530

youssef-lr committed:
Fix keyboard disabled on native

youssef-lr commented:
Can you please test the latest fix on iOS? I'm having issues with npm run ios

c3024 commented:
For `UGX` currency, the split does not reset correctly. It worked once correctly but failed every other time. I think there is some issue with currencies that cannot be split into decimals.


https://github.com/Expensify/App/assets/102477862/4fcb62f5-a37e-4e0a-aacd-f1401b33f949

youssef-lr committed:
Update MoneyRequestAmountInput.tsx

Co-authored-by: c3024 <102477862+c3024@users.noreply.github.com>

youssef-lr commented:
>For UGX currency, the split does not reset correctly. It worked once correctly but failed every other time. I think there is some issue with currencies that cannot be split into decimals.

Do we have the same issue on staging?

youssef-lr committed:
Rename callback

youssef-lr commented:
Looking into it, very weird bug.

youssef-lr commented:
wondering why does it work fine when I reset manually

https://github.com/Expensify/App/assets/9680864/b80b77bf-e548-4b2b-bfff-b7a983831f13

c3024 commented:
Tests well on iOS Native


https://github.com/Expensify/App/assets/102477862/5bac5f3f-15cf-427c-ac84-21d44f0cf327

c3024 commented:
Tests well on all platforms

Android

https://github.com/Expensify/App/assets/102477862/58518551-8c61-4c28-bba8-f82d4e52f12d

Android mWeb

https://github.com/Expensify/App/assets/102477862/c837186c-1075-45e1-af20-c88865c95278

iOS mWeb

https://github.com/Expensify/App/assets/102477862/ff85ba97-7c74-4ce4-b54d-13c5c4fddb69

Desktop

https://github.com/Expensify/App/assets/102477862/86e1fb6b-e509-404c-b170-e756a576d5b0

Chrome

https://github.com/Expensify/App/assets/102477862/0ab68add-aa75-4646-9b26-b23cae9cf0f7

dubielzyk-expensify commented:
Not sure if this is know, but I'm getting `Missing translation` on error:
![image](https://github.com/Expensify/App/assets/145416827/e4801fbf-cf0d-4952-b6c2-a9a72eb59a65)

dubielzyk-expensify commented:
Also, I'm not sure if this can be easily achieved, but if I press close to the edge (orange box in screenshot below) it doesn't register it as a tap on the input field, so it goes to my profile.
![CleanShot 2024-04-29 at 20 34 14@2x](https://github.com/Expensify/App/assets/145416827/7615a84f-c004-4edb-8ccc-6095d3df8771)

dubielzyk-expensify commented:
For accessibility reasons we should make sure the tap target on the `Reset` button is at least `44px x 44px`. I keep misclicking it at the moment. Something like this for a tap target would be good:
![CleanShot 2024-04-29 at 20 36 51@2x](https://github.com/Expensify/App/assets/145416827/26735333-98d8-4df8-97aa-4fb0cdecdac8)

dubielzyk-expensify commented:
Also, there's a tiny, but significant delay in updating the values of the split. It feels a bit jarring and jumpy when I change the values:

https://github.com/Expensify/App/assets/145416827/f9cad681-a82c-4f42-9ec9-6eb8e1fbca4b

When I go frame-by-frame it looks like the transition between two numbers are filled with `...`:
![CleanShot 2024-04-29 at 20 43 38@2x](https://github.com/Expensify/App/assets/145416827/46dfab88-a526-4f9a-905c-12bb6b900366)

I think ideally we'd want it to go from old number -> new number directly without an intermediary step. This is one where a "loading state" as such actually introduces more confusion, so I think we should remove it. Thoughts @youssef-lr ?


cc @Expensify/design

shawnborton commented:
Good catch Jon, I agree with that.

dannymcclain commented:
Yup, agree with that too Jon.

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
Bug fix

rafecolton reviewed (commented):
@youssef-lr can you please merge main? Also, any progress on sending the right IOU report IDs for delegated splits?

youssef-lr commented:
Ok I figured out the issue with the amount flickering, it's due to `autogrow`, when this prop is supplied we calculate the width of the input based on the number of characters. When we type a new character and given how the text has `text-align: left` and the container has `align-items: flex-end` (which is what allows us to right align the amount and the currency), the new character goes outside the input because the input still has the old width, once the new width is calculated it gets positioned correctly inside the input.

The alternative I found for now is to not align items using flex-end so that we'll have space to the right of digits. @ZhenjaHorbach also has found a way to make this work and we're troubleshooting it together as his solution has a padding that keeps getting added, here are the recordings of both solutions:

1. Removing align-items: flex-end


https://github.com/Expensify/App/assets/9680864/f6172efb-9517-4dfe-8b37-352d1fc3e34a

2. Same thing,  but adding `text-align: right` to the input, but then the currency will be far to the left

https://github.com/Expensify/App/assets/9680864/a2cf215a-8b89-4adc-98b9-7f3b75db8fc0

3. @ZhenjaHorbach's solution which is still in progress, we have to figure out the extra padding that keeps getting added


https://github.com/Expensify/App/assets/9680864/63442313-ae45-47ce-82a4-edf528334b23

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr committed:
TS fix

dubielzyk-expensify commented:
3 is very close to what we want though like you said we want to reduce the padding between the currency symbol and the text. 

If not, my personal take is that 2 feels better, but I don't feel super strongly. Keen to hear what @Expensify/design team thinks

shawnborton commented:
Yup, I'm aligned with Jon above.

youssef-lr commented:
@ZhenjaHorbach has figured out the bug in this commit https://github.com/Expensify/App/pull/41373/files, I'll merge that commit to my branch shortly so we can test again. I just need to figure out how to make the code easy to understand because it won't be clear to everybody what it's doing, I'll try adding clear comments.

youssef-lr committed:
Fix input flickering

youssef-lr committed:
Bug fixes and add logic to native

youssef-lr committed:
Style

youssef-lr committed:
Add autoGrowDirection

youssef-lr committed:
Cleanup

youssef-lr committed:
Add a missing change to native

youssef-lr committed:
Cleanup

youssef-lr committed:
TS

youssef-lr committed:
Bug fix

youssef-lr committed:
Show dot in numeric keyboard

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr commented:
Build was failing randomly, merged main and trying again.

rafecolton commented:
Looks like the Web and Auth PRs are both in prod so we can take this off hold

dubielzyk-expensify commented:
Still getting the `Missing translation` bug:
![CleanShot 2024-05-02 at 13 45 09@2x](https://github.com/Expensify/App/assets/145416827/625d0051-1637-4283-b50c-cb7365a5bcf8)

dubielzyk-expensify commented:
And the touch target on the reset button is still too small like mentioned [here](https://github.com/Expensify/App/pull/40386#issuecomment-2082389665)

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr commented:
>Still getting the Missing translation bug:

I honestly don't know why is this happening, the translations are already added. Could be an issue with the ad hoc build only?

youssef-lr committed:
Add accessibility hit slope

youssef-lr commented:
>And the touch target on the reset button is still too small like mentioned

Should be fixed now. I'm working on alternative solution #2 and will update the PR then create a build.

youssef-lr committed:
Revert implementation of autgrow with direction to the left

youssef-lr committed:
Revert unintended changes

youssef-lr commented:
>Still getting the Missing translation bug:

I've confirmed this is an issue with new translation keys on AdHoc builds only. This shouldn't be an issue once the PR is in staging or prod.

youssef-lr committed:
Don't format amount when text input is focused

youssef-lr committed:
Don't break other existing amount inputs

youssef-lr committed:
Remove unneeded prop

youssef-lr committed:
Remove unused param

youssef-lr committed:
Update styles and right-align input

youssef-lr committed:
Adjust padding based on prefix length relating to uppercase letters

youssef-lr committed:
Replace comma input with period for native keyboards that show a comma

dubielzyk-expensify commented:
Now there's two issues that hasn't appeared before

https://github.com/Expensify/App/assets/145416827/f361ecc7-c3ed-4a6d-bc3b-a7f5a1eed9f7

1) The footer button jumps up from the correct, to incorrect spacing at the bottom:
![CleanShot 2024-05-06 at 09 28 00](https://github.com/Expensify/App/assets/145416827/d3898bee-6eef-4184-b37e-1d97e9de4db9)

2) And during certain moments you can see the numbers jumping to two lines:
![CleanShot 2024-05-06 at 09 28 43@2x](https://github.com/Expensify/App/assets/145416827/18b8fd3f-7037-4b58-ad81-4dc9a52d9cc7)

youssef-lr committed:
Calculate exact width of the input based on total amount length

youssef-lr commented:
@dubielzyk-expensify that flicker seems to be unavoidable now if we use autogrow for the text input, there seems to be a delay when the amount gets bigger and we adjust the input width to accommodate for it, for a brief moment the input gets updated with the number while still having the old width. The contributor and I tried to figure this out with no luck.

youssef-lr committed:
Add callback removed by mistake

youssef-lr committed:
Allow editing formatted amount

youssef-lr commented:
@dubielzyk-expensify for the split button moving, it seems to be an issue on main and can be seen in the participants page as well, notice how the the 'Next' button moves a bit

https://github.com/Expensify/App/assets/9680864/cad9f3cd-eea2-46d1-9a51-933d78b870e4

youssef-lr committed:
Use fixed width for inputs and max length

youssef-lr committed:
Don't format amount on blur if it exceeds max length

youssef-lr committed:
Reset selection on blur on mobile

youssef-lr committed:
Merge branch 'main' into youssef_uneven_splits_2

youssef-lr commented:
@dubielzyk-expensify I've pushed a few fixes to eliminate the flicker in the inputs, let me know what you think!

dubielzyk-expensify reviewed (dismissed):
As for the split stuff this is looking awesome.

I'm still getting the jumpy Next button and Missing translation, but since you've mentioned this isn't related to this PR, I'll go ahead and approve. 

Thanks for going through all these details. It was worth it 😄

dubielzyk-expensify commented:
The "Paid by" appears to be sticky as well now, which is not correct

youssef-lr committed:
Fix paidBy section being sticky

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

youssef-lr commented:
PR above should fix this issue ![image](https://github.com/Expensify/App/assets/9680864/5f8b8c2b-3fd0-4ee8-8d27-884c9eff690a)

youssef-lr commented:
QA please don't check this off until the PR above is CP'ed.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9011216944) to production by https://github.com/marcaaron in version: 1.4.71-6 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

abdulrahuman5196 commented:
We had minor UI issue from this PR, more information here https://github.com/Expensify/App/issues/41751#issuecomment-2099742434

rushatgabhane commented on src/components/MoneyRequestConfirmationList.tsx:
```diff
@@ -616,6 +658,10 @@ function MoneyRequestConfirmationList({
                 return;
             }
 
+            if (formError) {
```
We caused a bug. RCA here - https://github.com/Expensify/App/issues/41336#issuecomment-2098318729

mollfpr commented on src/components/MoneyRequestConfirmationList.tsx:
```diff
@@ -709,14 +757,28 @@ function MoneyRequestConfirmationList({
                         style={[styles.ph1, styles.mb2]}
                         isError
-                        message={formError}
+                        message={!shouldShowReadOnlySplits ? debouncedFormError : formError}
```
Hi ✋ Coming from https://github.com/Expensify/App/issues/43184

The check `shouldShowReadOnlySplits` should be used on the split request, so adding `isTypeSplit` to avoid the `debouncedFormError` called on the non-split request.

rojiphil commented on src/components/MoneyRequestConfirmationList.tsx:
```diff
@@ -427,56 +415,121 @@ function MoneyRequestConfirmationList({
+                            ? IOUUtils.calculateAmount(selectedParticipants.length, iouAmount, iouCurrencyCode ?? '', isPayer)
+                            : transaction.comment.splits.find((split) => split.accountID === participantOption.accountID)?.amount;
+                }
+                return {
```
We did not prevent opening up the profile page for non-existing accounts thereby resulting in #42841

mollfpr commented on src/pages/iou/request/step/IOURequestStepAmount.tsx:
```diff
@@ -182,7 +182,8 @@ function IOURequestStepAmount({
             if (shouldSkipConfirmation) {
-                if (iouType === CONST.IOU.TYPE.SPLIT) {
+                // Only skip confirmation when the split is not configurable, for now Smartscanned splits cannot be configured
+                if (iouType === CONST.IOU.TYPE.SPLIT && transaction?.iouRequestType === CONST.IOU.REQUEST_TYPE.SCAN) {
```
✋ Coming from https://github.com/Expensify/App/issues/52386

On `MoneyRequestAmountForm`, the button text is because the `skipConfirmation` does not reflect the condition that IOU Manual Split can't use the skip confirmation.

We should update the flag `shouldSkipConfirmation` instead, so it can be reflected with the correct value in `MoneyRequestAmountForm`.

Eskalifer1 commented on src/libs/actions/IOU.ts:
```diff
@@ -6381,6 +6396,107 @@ function setShownHoldUseExplanation() {
+        };
+    }, {});
+
+    Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${transaction.transactionID}`, {splitShares});
```
When a user edited the Amount on a Split that had already been sent, a ghost draft transaction was created, which meant that when the next draft transaction was created, there was already one in existence. As a result, there were two draft transactions on the confirmation page, even though there should have been only one.

The reason is that `ONYXKEYS.COLLECTION.TRANSACTION_DRAFT` is always set here, regardless of whether `isDraft` is true or not.

Issue: https://github.com/Expensify/App/issues/88229
Fixed here: https://github.com/Expensify/App/pull/90716
